.github/workflows/validate.yml

```diff
@@ -60,28 +54,33 @@ env:
 
 jobs:
   validate:
-    name: Validate ${{ matrix.os }} ghc-${{ matrix.ghc }}
-    runs-on: ${{ matrix.os }}
+    name: Validate ${{ matrix.sys.os }} ghc-${{ matrix.ghc }}
+    runs-on: ${{ matrix.sys.os }}
     outputs:
       GHC_FOR_RELEASE: ${{ format('["{0}"]', env.GHC_FOR_RELEASE) }}
     strategy:
       matrix:
-        os: [ubuntu-latest, macos-13, windows-latest]
+        sys:
+          - { os: windows-latest, shell: 'C:/msys64/usr/bin/bash.exe -e {0}' }
+          - { os: ubuntu-latest,  shell: bash }
+          - { os: macos-13,       shell: bash}
         # If you remove something from here, then add it to the old-ghcs job.
         # Also a removed GHC from here means that we are actually dropping
         # support, so the PR *must* have a changelog entry.
         ghc: ['9.10.1', '9.8.2', '9.6.4', '9.4.8', '9.2.8', '9.0.2', '8.10.7', '8.8.4', '8.6.5']
         exclude:
           # corrupts GHA cache or the fabric of reality itself, see https://github.com/haskell/cabal/issues/8356
-          - os: windows-latest
+          - sys: { os: windows-latest, shell: 'C:/msys64/usr/bin/bash.exe -e {0}' }
```

Comment:
A little sad that you have to repeat this shell incantation in every include/exclude clause. Imagine this shell call will change a little... Are you positive that leaving out the `shell` field here is not acceptable?

Reply:
I tried to exclude by:
```
exclude:
  - sys.os: windows-latest
  - ghc: '8.10.7'
```
But it doesn't work :(

Reply:
Oh well, that's okay then. (I imagined that you did try it, and just wanted to double check.)

One day their workflow description syntax will support global constants...